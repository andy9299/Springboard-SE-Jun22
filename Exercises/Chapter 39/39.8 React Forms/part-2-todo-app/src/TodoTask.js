import React from "react";

function TodoTask({
  id,
  removeTask,
  task = "Empty Task!"
}) {
  const handleRemove = () => removeTask(id);
  return (
    <div>
      {task}
      <button onClick={handleRemove}>Remove Task</button>
    </div>
  );
}

export default TodoTask;
