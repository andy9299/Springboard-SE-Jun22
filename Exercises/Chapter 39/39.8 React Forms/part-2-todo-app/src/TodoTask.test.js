import React from 'react';
import { render } from "@testing-library/react";
import TodoTask from './TodoTask';

it("renders without crashing", function () {
  render(<TodoTask />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoTask />);
  expect(asFragment()).toMatchSnapshot();
});
