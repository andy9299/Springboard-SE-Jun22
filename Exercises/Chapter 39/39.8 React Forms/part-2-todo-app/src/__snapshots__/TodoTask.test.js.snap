// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Empty Task!
    <button>
      Remove Task
    </button>
  </div>
</DocumentFragment>
`;
